code/modules/projectiles/projectile.dm

```diff
@@ -177,7 +178,7 @@
 /obj/item/projectile/proc/on_impact(var/atom/A)
 	if (effect_transform)	//May be null when shooting something in the same tile
 		impact_effect(effect_transform)		// generate impact effect
-	if(damage && damage_type == BURN)
```

Comment:
nofire seems a bit hacky, it'd be better to add a projectile temperature value and utilize that instead of the hardcoded 700 here